src/auth/awsclient.ts

```diff
@@ -335,4 +337,21 @@ export class AwsClient extends BaseExternalAccountClient {
     });
     return response.data;
   }
+
+  private shouldUseMetadataServer(): boolean {
+    return (
+      !this.canRetrieveRegionFromEnvironment() ||
+      !this.canRetrieveSecurityCredentialsFromEnvironment()
+    );
+  }
+
+  private canRetrieveRegionFromEnvironment(): boolean {
+    return !!(process.env['AWS_REGION'] || process.env['AWS_DEFAULT_REGION']);
```

Comment:
nit: add same comment from java PR - "The AWS region can be provided through AWS_REGION or AWS_DEFAULT_REGION. Only one is required."